### PR TITLE
Random Beacon: "The Mechanism" README section updates

### LINKED ISSUE #2931: RandomBeacon contract mechanism documentation updates
We need to update *The Mechanism* documentation in [`solidity/random-beacon/README.md`](https://github.com/keep-network/keep-core/tree/main/solidity/random-beacon#the-mechanism) to reflect the most recent changes.

### PR DESCRIPTION
**Pull request by pdyraga**
Closes #2931

The section has been updated to reflect the most recent changes in smart contracts.

### CODE REVIEW

File: solidity/random-beacon/README.adoc
Comment (michalinacienciala) [inline solidity/random-beacon/README.adoc:60]: > One of the group members
Could we link that to the Operator-Only section in Transaction Incentives?

File: solidity/random-beacon/README.adoc
Comment (michalinacienciala) [inline solidity/random-beacon/README.adoc:70]: Maybe it would be worth to add that
- submitter of the successful challenge gets some percentage of the slashed amount
- after the result gets discarded the members of the group are given another chance to publish the result and DKG timeout timer is reset

File: solidity/random-beacon/README.adoc
Comment (michalinacienciala) [inline solidity/random-beacon/README.adoc:74]:
```suggestion
calling `RandomBeacon.approveDkgResult(DKG.Result calldata dkgResult)`.
```

File: solidity/random-beacon/README.adoc
Comment (michalinacienciala) [inline solidity/random-beacon/README.adoc:75]:
```suggestion
The submitter receives an ETH reimbursement for both `submitDkgResult` and
```

File: solidity/random-beacon/README.adoc
Comment (michalinacienciala) [inline solidity/random-beacon/README.adoc:88]:
```suggestion
DKG result to avoid front-running and minimize the cost, but no ordering is
```

File: solidity/random-beacon/README.adoc
Comment (michalinacienciala) [inline solidity/random-beacon/README.adoc:89]: Could we link 'submission order' to the Operator-Only section in Transaction Incentives?

File: solidity/random-beacon/README.adoc
Comment (michalinacienciala) [inline solidity/random-beacon/README.adoc:114]:
```suggestion
relay entry to avoid front-running and minimize the cost, but no ordering is
```

File: solidity/random-beacon/README.adoc
Comment (michalinacienciala) [inline solidity/random-beacon/README.adoc:115]: Could we link 'submission order' to the Operator-Only section in Transaction Incentives?

File: solidity/random-beacon/README.adoc
Comment (michalinacienciala) [inline solidity/random-beacon/README.adoc:167]: Maybe link 'notifier reward' (or earlier 'anyone can notify') to Punishment section in Transaction Incentives?

File: solidity/random-beacon/README.adoc
Comment (michalinacienciala) [inline solidity/random-beacon/README.adoc:178]:
```suggestion
This function unlocks the sortition pool and clears up DKG data, but no slashing
```

File: solidity/random-beacon/README.adoc
Comment (michalinacienciala) [inline solidity/random-beacon/README.adoc:198]: Could we link to Rewards section here?

File: solidity/random-beacon/README.adoc
Comment (michalinacienciala) [inline solidity/random-beacon/README.adoc:200]:
```suggestion
This approach is theoretically susceptible to group members colluding together,
```

File: solidity/random-beacon/README.adoc
Comment (michalinacienciala) [inline solidity/random-beacon/README.adoc:219]: The `[[inactivity]]` section describes only the situation when the `notifyOperatorInactivity()` function is called. The operators can be marked as ineligible for rewards also as a result of inactivity during off-chain DKG, when `approveDkgResult()` function is called, we should includa that case here.

File: solidity/random-beacon/README.adoc
Comment (michalinacienciala) [inline solidity/random-beacon/README.adoc:225]:
```suggestion
There are three types of transactions: <<operator-only,Operator-Only>>,
```

File: solidity/random-beacon/README.adoc
Comment (michalinacienciala) [inline solidity/random-beacon/README.adoc:240]: Link `inactive` to `[[inactivity]]`?

File: solidity/random-beacon/README.adoc
Comment (michalinacienciala) [inline solidity/random-beacon/README.adoc:246]: Could we mention about the cap for the reimbursed gas price? Or link to the RFC where it's described?
Same comment applies for the 'Public-Knowledge' section.

File: solidity/random-beacon/README.adoc
Comment (michalinacienciala) [inline solidity/random-beacon/README.adoc:249]: Well, `approveDkgResult` is not a tx where only group members have required data. Also, we don't follow the order of posting it based on the group's pubkey hash, instead we have one specific member who can post it (at least at the beginning) - but I understand why it's mentioned here and don't have idea how we could make the categorization more precise without making it too complicated...

File: solidity/random-beacon/README.adoc
Comment (michalinacienciala) [inline solidity/random-beacon/README.adoc:261]: We don't have the idea of 'trusted maintainers' in Random Beacon, right? Do we already have an idea how the clients will decide whose order it is to post the tx?

File: solidity/random-beacon/README.adoc
Comment (michalinacienciala) [inline solidity/random-beacon/README.adoc:269]: Do we have anything planned for preventing operators front-running each other to get the DKG submitter's reward?

File: solidity/random-beacon/README.adoc
Comment (michalinacienciala) [inline solidity/random-beacon/README.adoc:272]: Not sure where it would be the best place to put it, but we could mention somewhere what happens to the slashed amount - part goes as a reward, what happens to the other part?

**Review by michalinacienciala — COMMENTED** [review]
Left bunch of comments... Mostly small stuff...

File: solidity/random-beacon/README.adoc
Comment (pdyraga) [reply solidity/random-beacon/README.adoc]: 1fadd0bedacc1e8332f6ef3c24e1f6db4acc3322

File: solidity/random-beacon/README.adoc
Comment (pdyraga) [reply solidity/random-beacon/README.adoc]: 1fadd0bedacc1e8332f6ef3c24e1f6db4acc3322

File: solidity/random-beacon/README.adoc
Comment (pdyraga) [reply solidity/random-beacon/README.adoc:92]: 1fadd0bedacc1e8332f6ef3c24e1f6db4acc3322

File: solidity/random-beacon/README.adoc
Comment (pdyraga) [reply solidity/random-beacon/README.adoc]: 1fadd0bedacc1e8332f6ef3c24e1f6db4acc3322

File: solidity/random-beacon/README.adoc
Comment (pdyraga) [reply solidity/random-beacon/README.adoc]: 1fadd0bedacc1e8332f6ef3c24e1f6db4acc3322

File: solidity/random-beacon/README.adoc
Comment (pdyraga) [reply solidity/random-beacon/README.adoc]: 1fadd0bedacc1e8332f6ef3c24e1f6db4acc3322

File: solidity/random-beacon/README.adoc
Comment (pdyraga) [reply solidity/random-beacon/README.adoc]: 1fadd0bedacc1e8332f6ef3c24e1f6db4acc3322

File: solidity/random-beacon/README.adoc
Comment (pdyraga) [reply solidity/random-beacon/README.adoc]: cbd9a5f629ad089d45d2d66467c661e0eef2ba68

File: solidity/random-beacon/README.adoc
Comment (pdyraga) [reply solidity/random-beacon/README.adoc]: cbd9a5f629ad089d45d2d66467c661e0eef2ba68

File: solidity/random-beacon/README.adoc
Comment (pdyraga) [reply solidity/random-beacon/README.adoc]: cbd9a5f629ad089d45d2d66467c661e0eef2ba68

File: solidity/random-beacon/README.adoc
Comment (pdyraga) [reply solidity/random-beacon/README.adoc]: cbd9a5f629ad089d45d2d66467c661e0eef2ba68

File: solidity/random-beacon/README.adoc
Comment (pdyraga) [reply solidity/random-beacon/README.adoc]: cbd9a5f629ad089d45d2d66467c661e0eef2ba68

File: solidity/random-beacon/README.adoc
Comment (pdyraga) [reply solidity/random-beacon/README.adoc]: cbd9a5f629ad089d45d2d66467c661e0eef2ba68

File: solidity/random-beacon/README.adoc
Comment (pdyraga) [reply solidity/random-beacon/README.adoc:250]: Yeah. I explained it as "operator-only" for a certain number of blocks. I think it's good enough, and better than introducing a 4th category :)

File: solidity/random-beacon/README.adoc
Comment (pdyraga) [reply solidity/random-beacon/README.adoc:273]: It gets burned. I think it's a logic of a staking contract and we shouldn't talk about it. It is getting quite complicated when we go into details.

File: solidity/random-beacon/README.adoc
Comment (pdyraga) [reply solidity/random-beacon/README.adoc]: 07ad25c3b3bf9c7f2a23853a0b071397ee62a873

File: solidity/random-beacon/README.adoc
Comment (pdyraga) [reply solidity/random-beacon/README.adoc]: 07ad25c3b3bf9c7f2a23853a0b071397ee62a873

File: solidity/random-beacon/README.adoc
Comment (pdyraga) [reply solidity/random-beacon/README.adoc]: 07ad25c3b3bf9c7f2a23853a0b071397ee62a873

File: solidity/random-beacon/README.adoc
Comment (pdyraga) [reply solidity/random-beacon/README.adoc:267]: We do not need them for Random Beacon. The only "maintainer" transaction is `notifyDkgTimeOut` and it should not be called often (hopefully never called), so we do not have to worry about the order of transaction submitters. In tBTC, we have submit sweep proof, submit redemption proof that are called *very often* and there, we need to worry about the order maintainers should follow.

File: solidity/random-beacon/README.adoc
Comment (pdyraga) [reply solidity/random-beacon/README.adoc:275]: I do not think we need it. In the vast majority of cases, the original submitter should approve - otherwise, it does not make sense to submit.

**Comment by pdyraga** [comment]
Alright @michalinacienciala, all comments addressed, this is ready for another chance!

File: solidity/random-beacon/README.adoc
Comment (michalinacienciala) [reply solidity/random-beacon/README.adoc:255]: Yeah, let's leave it as it is.

File: solidity/random-beacon/README.adoc
Comment (michalinacienciala) [reply solidity/random-beacon/README.adoc]: Could we add that as well to the 'Public-Knowledge' section? Can be something like ("again, limited by the gas price cap")...

File: solidity/random-beacon/README.adoc
Comment (michalinacienciala) [reply solidity/random-beacon/README.adoc:255]: Yeah, let's leave it as it is.

File: solidity/random-beacon/README.adoc
Comment (michalinacienciala) [reply solidity/random-beacon/README.adoc:278]: OK, let's leave it as it is.

**Review by michalinacienciala — APPROVED** [review]
Left one comment, optional. If you think we should left it as it is, I'm ok with merging this PR.

File: solidity/random-beacon/README.adoc
Comment (pdyraga) [reply solidity/random-beacon/README.adoc]: I intentionally left it only in this section to not complicate the description of the second section too much. It feels to me that explaining it once is enough. WDYT?

Comment (michalinacienciala) [reply solidity/random-beacon/README.adoc]: OK, I'm fine with leaving it as it is

**Review by michalinacienciala — APPROVED** [review]

**Pull request merged**
